internal/core/command/put_test.go

```diff
@@ -73,3 +82,72 @@ func TestNewPutCommandBody(t *testing.T) {
 		t.Error("Failed to verify the request body contents")
 	}
 }
+
+func TestNewPutCommandContentType(t *testing.T) {
+	tests := []struct {
+		name                string
+		expectedContentType string
+		req                 *http.Request
+	}{
+		{
+			"cbor content type",
+			clients.ContentTypeCBOR,
+			newRequestForPutWithContentType(clients.ContentTypeCBOR),
+		},
+		{
+			"json content type",
+			clients.ContentTypeJSON,
+			newRequestForPutWithContentType(clients.ContentTypeJSON),
+		},
+		{
+			"no content type was provided",
+			"",
+			newRequestForPutWithContentType(""),
+		},
+		{
+			"user-agent header should not be propagated",
+			"",
+			newRequestForPutWithUnexpectedUserAgentHeader("User-Agent"),
+		},
+		{
+			"content-md5 header should not be propagated",
+			"",
+			newRequestForPutWithUnexpectedContentMD5Header("Content-MD5"),
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			var loggerMock = logger.NewMockClient()
+			ctx := context.Background()
+			putCommand, _ := NewPutCommand(
+				testDevice,
+				testCommand,
+				"Test Body",
+				ctx,
+				nil,
+				loggerMock,
+				tt.req)
+			actualContentType := putCommand.(serviceCommand).Request.Header.Get(clients.ContentType)
+			require.Equal(t, tt.expectedContentType, actualContentType)
+		})
+	}
+}
+
+func newRequestForPutWithContentType(contentType string) *http.Request {
+	req, _ := http.NewRequest(http.MethodGet, "/", strings.NewReader("Test body"))
+	req.Header.Set(clients.ContentType, contentType)
+	return req
+}
+
+func newRequestForPutWithUnexpectedUserAgentHeader(headerType string) *http.Request {
+	req, _ := http.NewRequest(http.MethodGet, "/", strings.NewReader("Test body"))
+	req.Header.Set("User-Agent", headerType)
+	return req
+}
+
+func newRequestForPutWithUnexpectedContentMD5Header(headerType string) *http.Request {
+	req, _ := http.NewRequest(http.MethodGet, "/", strings.NewReader("Test body"))
+	req.Header.Set("Content-MD5", headerType)
+	return req
+}
```

Comment:
Please see my previous comment regarding handling all these cases in one more general function.

Also, there is code duplication between this file and `get_test.go` can we move this logic to a common place and leverage that to eliminate code duplication?